```
pymc/tests/test_distributions.py:1943: 
519
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
520
pymc/tests/test_distributions.py:705: in check_logp
521
    model, param_vars = build_model(pymc_dist, domain, paramdomains, extra_args)
522
pymc/tests/test_distributions.py:282: in build_model
523
    distfam(
524
pymc/distributions/distribution.py:255: in __new__
525
    rv_out, dims, observed, resize_shape = _make_rv_and_resize_shape(
526
pymc/distributions/distribution.py:154: in _make_rv_and_resize_shape
527
    rv_out = cls.dist(*args, **kwargs)
528
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
529

530
cls = <class 'pymc.distributions.multivariate.MatrixNormal'>, mu = mu
531
rowcov = rowcov, rowchol = None, colcov = colcov, colchol = None, args = ()
532
kwargs = {'rng': RandomStateSharedVariable(<RandomState(MT19937) at 0x7FEA3124F340>)}
533
cholesky = <aesara.tensor.slinalg.Cholesky object at 0x7fea305062e0>
534
rowchol_cov = Cholesky{lower=True, destructive=False, on_error='raise'}.0
535
colchol_cov = Cholesky{lower=True, destructive=False, on_error='raise'}.0
536
dist_shape = (Subtensor{int64}.0, Subtensor{int64}.0)
537

538
    @classmethod
539
    def dist(
540
        cls,
541
        mu,
542
        rowcov=None,
543
        rowchol=None,
544
        colcov=None,
545
        colchol=None,
546
        *args,
547
        **kwargs,
548
    ):
549
    
550
        cholesky = Cholesky(lower=True, on_error="raise")
551
    
552
        if kwargs.get("size", None) is not None:
553
            raise NotImplementedError("MatrixNormal doesn't support size argument")
554
    
555
        if "shape" in kwargs:
556
            kwargs.pop("shape")
557
            warnings.warn(
558
                "The shape argument in MatrixNormal is deprecated and will be ignored."
559
                "MatrixNormal automatically derives the shape"
560
                "from row and column matrix dimensions.",
561
                FutureWarning,
562
            )
563
    
564
        # Among-row matrices
565
        if len([i for i in [rowcov, rowchol] if i is not None]) != 1:
566
            raise ValueError(
567
                "Incompatible parameterization. Specify exactly one of rowcov, or rowchol."
568
            )
569
        if rowcov is not None:
570
            if rowcov.ndim != 2:
571
                raise ValueError("rowcov must be two dimensional.")
572
            rowchol_cov = cholesky(rowcov)
573
        else:
574
            if rowchol.ndim != 2:
575
                raise ValueError("rowchol must be two dimensional.")
576
            rowchol_cov = at.as_tensor_variable(rowchol)
577
    
578
        # Among-column matrices
579
        if len([i for i in [colcov, colchol] if i is not None]) != 1:
580
            raise ValueError(
581
                "Incompatible parameterization. Specify exactly one of colcov, or colchol."
582
            )
583
        if colcov is not None:
584
            colcov = at.as_tensor_variable(colcov)
585
            if colcov.ndim != 2:
586
                raise ValueError("colcov must be two dimensional.")
587
            colchol_cov = cholesky(colcov)
588
        else:
589
            if colchol.ndim != 2:
590
                raise ValueError("colchol must be two dimensional.")
591
            colchol_cov = at.as_tensor_variable(colchol)
592
    
593
        dist_shape = (rowchol_cov.shape[0], colchol_cov.shape[0])
594
    
595
        # Broadcasting mu
596
>       mu = at.extra_ops.broadcast_to(a=mu, shape=dist_shape)
597
E       TypeError: broadcast_to() got an unexpected keyword argument 'a'
598

599
pymc/distributions/multivariate.py:1745: TypeError
```

--- a/conda-envs/environment-dev-py37.yml
+++ b/conda-envs/environment-dev-py37.yml
@@ -5,7 +5,7 @@ channels:
 - defaults
 dependencies:
 - aeppl=0.0.18
-- aesara=2.3.2
+- aesara=2.3.6
 - arviz>=0.11.4
 - cachetools>=4.2.1
 - cloudpickle

--- a/conda-envs/environment-dev-py37.yml
+++ b/conda-envs/environment-dev-py37.yml
@@ -13,7 +13,7 @@ dependencies:
 - h5py>=2.7
 - ipython
 - myst-nb
-- numpy>=1.15.0,<1.22.0
+- numpy>=1.15
 - numpydoc>=0.9
 - pandas>=0.24
 - pre-commit>=2.8.0

--- a/conda-envs/environment-dev-py38.yml
+++ b/conda-envs/environment-dev-py38.yml
@@ -5,7 +5,7 @@ channels:
 - defaults
 dependencies:
 - aeppl=0.0.18
-- aesara=2.3.2
+- aesara=2.3.6
 - arviz>=0.11.4
 - cachetools>=4.2.1
 - cloudpickle

--- a/conda-envs/environment-dev-py38.yml
+++ b/conda-envs/environment-dev-py38.yml
@@ -13,7 +13,7 @@ dependencies:
 - h5py>=2.7
 - ipython
 - myst-nb
-- numpy>=1.15.0,<1.22.0
+- numpy>=1.15.0
 - numpydoc>=0.9
 - pandas>=0.24.0
 - pre-commit>=2.8.0

--- a/conda-envs/environment-dev-py39.yml
+++ b/conda-envs/environment-dev-py39.yml
@@ -5,7 +5,7 @@ channels:
 - defaults
 dependencies:
 - aeppl=0.0.18
-- aesara=2.3.2
+- aesara=2.3.6
 - arviz>=0.11.4
 - cachetools>=4.2.1
 - cloudpickle

--- a/conda-envs/environment-dev-py39.yml
+++ b/conda-envs/environment-dev-py39.yml
@@ -13,7 +13,7 @@ dependencies:
 - h5py>=2.7
 - ipython
 - myst-nb
-- numpy>=1.15.0,<1.22.0
+- numpy>=1.15.0
 - numpydoc>=0.9
 - pandas
 - pre-commit>=2.8.0

--- a/conda-envs/environment-test-py37.yml
+++ b/conda-envs/environment-test-py37.yml
@@ -5,7 +5,7 @@ channels:
 - defaults
 dependencies:
 - aeppl=0.0.18
-- aesara=2.3.2
+- aesara=2.3.6
 - arviz>=0.11.4
 - cachetools>=4.2.1
 - cloudpickle

--- a/conda-envs/environment-test-py37.yml
+++ b/conda-envs/environment-test-py37.yml
@@ -14,7 +14,7 @@ dependencies:
 - ipython
 - libblas=*=*mkl
 - mkl-service
-- numpy>=1.15.0,<1.22.0
+- numpy>=1.15.0
 - pandas>=0.24
 - pre-commit>=2.8.0
 - pytest-cov>=2.5

--- a/conda-envs/environment-test-py38.yml
+++ b/conda-envs/environment-test-py38.yml
@@ -14,7 +14,7 @@ dependencies:
 - ipython
 - libblas=*=*mkl
 - mkl-service
-- numpy>=1.15.0,<1.22.0
+- numpy>=1.15.0
 - pandas
 - pre-commit>=2.8.0
 - pytest-cov>=2.5

--- a/conda-envs/environment-test-py38.yml
+++ b/conda-envs/environment-test-py38.yml
@@ -5,7 +5,7 @@ channels:
 - defaults
 dependencies:
 - aeppl=0.0.18
-- aesara=2.3.2
+- aesara=2.3.6
 - arviz>=0.11.4
 - cachetools>=4.2.1
 - cloudpickle

--- a/conda-envs/environment-test-py39.yml
+++ b/conda-envs/environment-test-py39.yml
@@ -5,7 +5,7 @@ channels:
 - defaults
 dependencies:
 - aeppl=0.0.18
-- aesara=2.3.2
+- aesara=2.3.6
 - arviz>=0.11.4
 - cachetools
 - cloudpickle

--- a/conda-envs/environment-test-py39.yml
+++ b/conda-envs/environment-test-py39.yml
@@ -14,7 +14,7 @@ dependencies:
 - ipython>=7.16
 - libblas=*=*mkl
 - mkl-service
-- numpy>=1.15.0,<1.22.0
+- numpy>=1.15.0
 - pandas
 - pre-commit>=2.8.0
 - pytest-cov>=2.5

--- a/conda-envs/windows-environment-dev-py38.yml
+++ b/conda-envs/windows-environment-dev-py38.yml
@@ -11,7 +11,7 @@ dependencies:
 - cloudpickle
 - fastprogress>=0.2.0
 - h5py
-- numpy>=1.15.0,<1.22.0
+- numpy>=1.15.0
 - pandas
 - pip
 - python=3.8

--- a/conda-envs/windows-environment-dev-py38.yml
+++ b/conda-envs/windows-environment-dev-py38.yml
@@ -5,7 +5,7 @@ channels:
 dependencies:
  # base dependencies (see install guide for Windows)
 - aeppl=0.0.18
-- aesara=2.3.2
+- aesara=2.3.6
 - arviz>=0.11.4
 - cachetools>=4.2.1
 - cloudpickle

--- a/conda-envs/windows-environment-test-py38.yml
+++ b/conda-envs/windows-environment-test-py38.yml
@@ -5,7 +5,7 @@ channels:
 dependencies:
  # base dependencies (see install guide for Windows)
 - aeppl=0.0.18
-- aesara=2.3.2
+- aesara=2.3.6
 - arviz>=0.11.2
 - cachetools
 - cloudpickle

--- a/conda-envs/windows-environment-test-py38.yml
+++ b/conda-envs/windows-environment-test-py38.yml
@@ -15,7 +15,7 @@ dependencies:
 - mkl==2020.4
 - mkl-service==2.3.0
 - m2w64-toolchain
-- numpy>=1.15.0,<1.22.0
+- numpy>=1.15.0
 - pandas
 - pip
 - python=3.8

--- a/pymc/distributions/multivariate.py
+++ b/pymc/distributions/multivariate.py
@@ -1742,7 +1742,7 @@ class MatrixNormal(Continuous):
         dist_shape = (rowchol_cov.shape[0], colchol_cov.shape[0])
 
         # Broadcasting mu
-        mu = at.extra_ops.broadcast_to(a=mu, shape=dist_shape)
+        mu = at.extra_ops.broadcast_to(mu, shape=dist_shape)
 
         mu = at.as_tensor_variable(floatX(mu))
         # mean = median = mode = mu

--- a/pymc/ode/ode.py
+++ b/pymc/ode/ode.py
@@ -158,7 +158,7 @@ class DifferentialEquation(Op):
         theta = at.cast(at.unbroadcast(at.as_tensor_variable(theta), 0), floatX)
         inputs = [y0, theta]
         for i, (input_val, itype) in enumerate(zip(inputs, self._itypes)):
-            if not input_val.type == itype:
+            if not input_val.type.in_same_class(itype):
                 raise ValueError(
                     f"Input {i} of type {input_val.type} does not have the expected type of {itype}"
                 )

--- a/pymc/tests/test_smc.py
+++ b/pymc/tests/test_smc.py
@@ -140,20 +140,13 @@ class TestSMC(SeededTest):
             with pm.Model() as model:
                 a = pm.Beta("a", alpha, beta)
                 y = pm.Bernoulli("y", a, observed=data)
-                trace = pm.sample_smc(2000, return_inferencedata=False)
-                marginals.append(trace.report.log_marginal_likelihood)
+                trace = pm.sample_smc(2000, chains=2, return_inferencedata=False)
+            # log_marignal_likelihood is found in the last value of each chain
+            lml = np.mean([chain[-1] for chain in trace.report.log_marginal_likelihood])
+            marginals.append(lml)
 
         # compare to the analytical result
-        assert (
-            np.abs(
-                np.exp(
-                    np.nanmean(np.array(marginals[1], dtype=float))
-                    - np.nanmean(np.array(marginals[0], dtype=float))
-                    - 4.0
-                )
-            )
-            <= 1
-        )
+        assert abs(np.exp(marginals[1] - marginals[0]) - 4.0) <= 1
 
     def test_start(self):
         with pm.Model() as model:

--- a/requirements-dev.txt
+++ b/requirements-dev.txt
@@ -2,7 +2,7 @@
 # See that file for comments about the need/usage of each dependency.
 
 aeppl==0.0.18
-aesara==2.3.2
+aesara==2.3.6
 arviz>=0.11.4
 cachetools>=4.2.1
 cloudpickle

--- a/requirements-dev.txt
+++ b/requirements-dev.txt
@@ -10,7 +10,7 @@ fastprogress>=0.2.0
 h5py>=2.7
 ipython
 myst-nb
-numpy>=1.15.0,<1.22.0
+numpy>=1.15.0
 numpydoc>=0.9
 pandas
 pre-commit>=2.8.0

--- a/requirements.txt
+++ b/requirements.txt
@@ -4,7 +4,7 @@ arviz>=0.11.4
 cachetools>=4.2.1
 cloudpickle
 fastprogress>=0.2.0
-numpy>=1.15.0,<1.22.0
+numpy>=1.15.0
 pandas>=0.24.0
 scipy>=1.4.1
 typing-extensions>=3.7.4

--- a/requirements.txt
+++ b/requirements.txt
@@ -1,5 +1,5 @@
 aeppl==0.0.18
-aesara==2.3.2
+aesara==2.3.6
 arviz>=0.11.4
 cachetools>=4.2.1
 cloudpickle

